### PR TITLE
[ETFE-2276] - EIS integration for RoR

### PR DESCRIPTION
**Pull request by LewisMitchell**

### CODE REVIEW

**Comment by LewisMitchell** [comment]
Stub (running on ETFE-2277) returns OK (when SendToEIS feature switch is enabled). Frontend displays an error (because of a different response to ChRIS)

**Comment by LewisMitchell** [comment]
rebuild

File: app/uk/gov/hmrc/emcstfe/config/AppConfig.scala [hunk 1]
```diff
@@ -57,7 +59,9 @@ class AppConfig @Inject()(servicesConfig: ServicesConfig, configuration: Configu
 
   def getFeatureSwitchValue(feature: String): Boolean = configuration.get[Boolean](feature)
 
-  def chrisBaseUrl: String = if(isEnabled(UseChrisStub)) chrisStubUrl else chrisUrl
+  def chrisBaseUrl: String = if(isEnabled(UseChrisStub)) downstreamStubUrl else chrisUrl
```
Comment (MJCallahanPage) [inline app/uk/gov/hmrc/emcstfe/config/AppConfig.scala:62]: Wouldn't we just call this `isEnabled(DownstreamStub)`? I don't think we'd need two separate flags as it's the one dynamic stub that caters for both.

File: app/uk/gov/hmrc/emcstfe/featureswitch/core/config/FeatureSwitchingModule.scala [hunk 2]
```diff
@@ -39,3 +39,13 @@ case object UseChrisStub extends FeatureSwitch {
   override val displayName: String = "enables chris stub"
 }
 
+case object UseEisStub extends FeatureSwitch {
+  override val configName: String = "features.eisStub"
+  override val displayName: String = "Enables EIS stub"
+}
```
Comment (MJCallahanPage) [inline app/uk/gov/hmrc/emcstfe/featureswitch/core/config/FeatureSwitchingModule.scala:45]: Don't think we need this. Just rename the ChRIS Stub to `DownstreamStub`

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
Left a couple of small comments.

File: app/uk/gov/hmrc/emcstfe/controllers/SubmitReportOfReceiptController.scala [hunk 3]
```diff
@@ -30,15 +33,22 @@ import scala.concurrent.ExecutionContext
...
 
   def submit(ern: String, arc: String): Action[JsValue] = authorisedUserSubmissionRequest(ern) { implicit request =>
     withJsonBody[SubmitReportOfReceiptModel] { submission =>
-      service.submit(submission).map {
-        case Left(value) => InternalServerError(Json.toJson(value))
-        case Right(value) => Ok(Json.toJson(value))
+      if(isEnabled(SendToEIS)) {
+        service.submitViaEIS(submission).map(handleResponse(_))
+      } else {
+        service.submit(submission).map(handleResponse(_))
```
Comment (MJCallahanPage) [inline app/uk/gov/hmrc/emcstfe/controllers/SubmitReportOfReceiptController.scala:45]: Might be better to FS in the Service layer. As it's that which decides which connector to then use. Controller doesn't really need to care as method signature and response is the same.

File: app/uk/gov/hmrc/emcstfe/connectors/BaseEisConnector.scala [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+trait BaseEisConnector extends Logging {
+
+  def metricsService: MetricsService
+
+  private def eisHeaders(correlationId: String): Seq[(String, String)] = Seq(
+    EisHeaders.dateTime -> s"${Instant.now.truncatedTo(ChronoUnit.MILLIS)}",
+    EisHeaders.correlationId -> correlationId,
+    EisHeaders.forwardedHost -> "MDTP",
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/uk/gov/hmrc/emcstfe/connectors/BaseEisConnector.scala:36]: I reached out and asked, it seems a few other services (not in our team) use MDTP as the forwarded host, I could put this in app conf so it's easy to change in the future?

File: app/uk/gov/hmrc/emcstfe/controllers/SubmitReportOfReceiptController.scala (hunk 3 above)
Comment (LewisMitchell) [reply app/uk/gov/hmrc/emcstfe/controllers/SubmitReportOfReceiptController.scala:45]: Response wasn't the same :( my mistake. Kept as is

File: app/uk/gov/hmrc/emcstfe/connectors/BaseEisConnector.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (MJCallahanPage) [reply app/uk/gov/hmrc/emcstfe/connectors/BaseEisConnector.scala]: Sounds sensible.

**Review by MJCallahanPage — APPROVED** [review]
LGTM 🟢 🚀

**Pull request merged**